src/core/internal/dassert.d

```diff
@@ -88,7 +88,8 @@ private string miniFormat(V)(const ref V v)
     else static if (__traits(isFloating, V))
     {
         char[60] val;
-        const len = sprintf(&val[0], "%g", v);
+        const format = "%g"; // escape printf format validation
+        const len = sprintf(&val[0], format.ptr, v);
```

Comment:
That's not a fix; for C varargs, `float` is promoted to double, but D `real` as C `long double` needs a `%Lg` format...

Reply:
Fixed. However, why `cfloat` aren't promoted to `creal` in this case ?

Reply:
AFAIK, there's no complex type support for printf, and no promotions are done (it's seen as a struct with 2 fp values).

Reply:
I added support for complex numbers in checkPrintFormat (see https://github.com/dlang/dmd/pull/10872/commits/090c02012f063b98bcd06f1d64759f8efe32e893).
Should we stay with that ?

Reply:
According to https://stackoverflow.com/questions/4099433/c-complex-number-and-printf, printf doesn't support it.

Reply:
Then, what should I do with `V` in `miniFormat`, which sometimes is a `creal`, a `cdouble` or a `cfloat` ?

Reply:
Be creative - e.g., `"%[L]g + %[L]gi", v.re, v.im`.

Reply:
I see.